src/views/ProfilePage.vue

```diff
@@ -13,43 +17,142 @@
-          <user-avatars-list v-if="userProfile" :users="userProfile.friends">
-          </user-avatars-list>
+        <ion-item button id="open-modal" expand="block">
+          <ion-label>{{ $t("friends") }}</ion-label>
+          <user-avatars-list
+            v-if="userProfile"
+            :users="userProfile.friends"
+          ></user-avatars-list>
+          <ion-modal ref="modal" trigger="open-modal">
+            <ion-page>
+              <ion-header>
+                <ion-toolbar>
+                  <ion-buttons slot="start">
+                    <ion-button @click="cancel">Cancel</ion-button>
+                  </ion-buttons>
+                  <ion-title>{{ $t("friends") }}</ion-title>
+                </ion-toolbar>
+              </ion-header>
+              <ion-content class="ion-padding">
+                <ion-list :inset="true">
+                  <ion-list-header>
+                    <ion-label>{{ $t("friends") }}</ion-label>
+                  </ion-list-header>
+                  <div v-if="userProfile?.friends?.length > 0">
+                    <ion-item-sliding
+                      v-for="(user, idx) in userProfile?.friends"
+                      :key="user?.id"
+                    >
+                      <ion-item>
+                        <div slot="start" v-if="user?.image">
+                          <user-avatar
+                            class="small avatar"
+                            :userProfile="user"
+                          />
+                        </div>
+                        <ion-label>{{ user?.name }}</ion-label>
+                      </ion-item>
+                      <ion-item-options>
+                        <ion-item-option
+                          color="danger"
+                          @click="removeUser(user.id)"
+                          >{{ $t("delete") }}</ion-item-option
+                        >
+                      </ion-item-options>
+                    </ion-item-sliding>
+                  </div>
+                  <div v-else>
+                    {{ $t("noFriends") }}
+                  </div>
+                </ion-list>
+                <friends-request-list :currentUserDoc="currentUserDoc" />
+                <ion-fab>
+                  <ion-fab-button id="modal-friends" expand="block">
+                    <ion-icon :icon="add"></ion-icon>
+                  </ion-fab-button>
+                </ion-fab>
+                <ion-modal ref="modal-friends" trigger="modal-friends">
+                  <ion-page>
+                    <ion-header>
+                      <ion-toolbar>
+                        <ion-buttons slot="start">
+                          <ion-button @click="cancel">Cancel</ion-button>
+                        </ion-buttons>
+                        <ion-title>{{ $t("addFriends") }}</ion-title>
+                      </ion-toolbar>
+                    </ion-header>
+                  </ion-page>
+                </ion-modal>
```

Comment:
Suggestion: We should create components for every modal. This makes the code much more readable.

Reply:
we agree to take this as is, let's refactor this later.